app/components/documents/document-details-panel.hbs

```diff
@@ -0,0 +1,175 @@
+      {{/if}}
+      <Auk::KeyValue data-test-details-tab-name @key={{t "document-name"}}>
+        <Textarea
+          data-test-details-tab-edit-name
+          aria-labelledby="piece-name"
+          class="auk-textarea"
+          @value={{mut @piece.name}}
+        />
+      </Auk::KeyValue>
+      <Auk::KeyValue data-test-details-tab-access-level @key={{t "publicity"}}>
+        {{#if this.isEditingDetails}}
+          <Utils::AccessLevelSelector
+            @allowClear={{true}}
+            @displayField="label"
+            @sortField="position"
+            @selected={{this.accessLevel}}
+            @onChange={{this.setAccessLevel}}
+            data-test-details-tab-edit-access-level
+          />
+        {{else}}
+          <Utils::AccessLevelColorBadge @accessLevel={{this.accessLevel}} />
+        {{/if}}
+      </Auk::KeyValue>
+      <Auk::KeyValue
+        data-test-details-tab-access-level
+        @key={{t "source-file"}}
+      >
+        <p>
+          {{#if this.replacementFile}}
+            {{this.replacementFile.name}}
+          {{else}}
+            {{! Note that showing this composed name as "source file name" might be confusing.
+            Especially so when one just replaced the source and the same generated name pops up again (comoposed of piece name, which stays the same).
+            Better show original source file (not document) name?}}
+            {{@piece.name}}.{{@piece.file.extension}}
+          {{/if}}
+        </p>
+        <AuButton
+          @icon="upload"
+          @iconAlignment="left"
+          @skin="link"
+          {{on "click" (fn (mut this.isUploadingReplacementFile) true)}}
+        >{{t "replace"}}</AuButton>
+      </Auk::KeyValue>
+    </Auk::Panel::Body>
+    <Auk::Panel::Footer>
+      <Auk::Toolbar>
+        <Auk::Toolbar::Group @position="left">
+          <Auk::Toolbar::Item>
+            <Auk::Button
+              data-test-details-tab-cancel
+              @skin="borderless"
+              @disabled={{this.isProcessing}}
+              {{on "click" (perform this.cancelEditDetails)}}
+            >
+              {{t "cancel"}}
+            </Auk::Button>
+          </Auk::Toolbar::Item>
+        </Auk::Toolbar::Group>
+        <Auk::Toolbar::Group @position="right">
+          <Auk::Toolbar::Item>
+            <Auk::Button
+              data-test-details-tab-save
+              @skin="primary"
+              @disabled={{this.isProcessing}}
+              @loading={{this.saveDetails.isRunning}}
+              {{on "click" (perform this.saveDetails)}}
+            >
+              {{t "save"}}
+            </Auk::Button>
+          </Auk::Toolbar::Item>
+        </Auk::Toolbar::Group>
+      </Auk::Toolbar>
+    </Auk::Panel::Footer>
+  </Auk::Panel>
+  {{#if this.isUploadingReplacementFile}}
+    <Auk::Modal>
+      <Auk::Modal::Body>
+        <Auk::FileUploader @multiple={{false}} @onUpload={{this.replaceFile}} />
+      </Auk::Modal::Body>
+    </Auk::Modal>
+  {{/if}}
```

Comment:
Vervangen van de file zelf staat niet in de scope van het huidige ticket.
Dat wil niet zeggen dat het niet mag, maar dan passen we best de scope van het ticket aan zodat de gebruiker hier weet van hebben + beseffen dat het getest moet worden.

Ik heb daar wel nog een reeks vragen bij die feature.
Mag dit altijd vervangen worden? Wat als er de documenten al vrijgegeven zijn? Valvas heeft dan nog oude documenten en krijgt geen update? + meer...
Ik vermoed dat er een check mist 'canReplaceFiles' met wat logica erin.
Mogelijk willen we die feature even in comment zetten/vervolgticket  zodat de PR kan doorgaan.

Reply:
Ben wel voorstander om die replacement voorlopig nog uit deze PR te houden (of in comments) en in aparte story uit te werken. 

Logica voor `canReplaceFiles` moet inderdaad nog uitgeklaard worden. Voor doorstroming naar Valvas is de afspraak momenteel dat ze manueel re-export moeten triggeren indien ze wijzigingen maken, dus daar zie ik niet onmiddelijk een probleem. Voor doorstroming naar andere profielen ben ik niet zeker of Yggdrasil momenteel de oude file correct gaat opkuisen uit de andere graphs.

Reply:
> Vervangen van de file zelf staat niet in de scope van het huidige ticket. Dat wil niet zeggen dat het niet mag, maar dan passen we best de scope van het ticket aan zodat de gebruikers hier weet van hebben + beseffen dat het getest moet worden.
> 
> Ik heb daar wel nog een reeks vragen bij die feature. Mag dit altijd vervangen worden? Wat als er de documenten al vrijgegeven zijn? Valvas heeft dan nog oude documenten en krijgt geen update? + meer... Ik vermoed dat er een check mist 'canReplaceFiles' met wat logica erin. Mogelijk willen we die feature even in comment zetten/vervolgticket zodat de PR kan doorgaan.

@ValenberghsSven @erikap Als worddocumenten nog niet automatisch worden omgezet naar PDF is het ok om het vervangen van bestanden voorlopig uit deze scope te halen.

Reply:
Aangepast in https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1422/commits/486c37d33742aedc6dde877b322d93c893e59b4a